docs/source/troubleshooting.md

```diff
@@ -0,0 +1,85 @@
+- failures induced by environment problems, like misconfigured firewalls preventing nodes connecting to each others
+- transient consensus failures on some nodes, most likely to some unhandled edge cases, which can go away after restarting affected nodes
+- node failures, like inability to properly perform an upgrade or handle some incoming request due to some bug, which require manual intervention, but doesn't affect ledger data
+- ledger corruption, which require touching ledger data in order to fix it
+
+Of course, if number of affected nodes is f or less then from external point of view functionality of pool will be unaffected.
+However if more than f nodes become affected then pool will not be able to do writes, although pool still will be able to process reads until all nodes fail.
+
+### Where to get info
+
+Most useful places get info are the following:
+- either VALIDATOR_INFO command sent through Indy CLI, or `validator-info` script run on validator node. 
+  These tools provide important information like how many nodes are connected to each other, when last write happened (due to freshness check it should happen at least once per 5 minutes), whether a view change is in progress now or other important data, which is useful to assess pool current health and can be a starting point for further investigation, when needed.
+- `journalctl` logs can be useful because they contain tracebacks of indy-node crashes, if they happened, and these logs are really easy to check. 
+  Sometime crashes can be due to some bugs in code, but also they can be caused by insufficient resource (either memory or disk space), and if this is the case `journalctl` logs can provide a quick answer.
+- indy node logs, located in `/var/log/indy/<network name>/`. 
+  They can provide a lot of historical information, and very often are enough to get enough clues in order to properly diagnose situation, however they can be hard to read for unprepared.
+  Lately most of the time it was enough to use `grep` and `sort` command-line tools to analyze them (although we'd recommend using [ripgrep](https://github.com/BurntSushi/ripgrep/releases) instead of plain grep, as it has quite a bunch of usability and performance improvements over traditional grep, while having compatible interface and no extra dependencies).
+  However we also have a [process_logs](https://github.com/hyperledger/indy-plenum/tree/master/scripts/process_logs) utility script, which also can be useful.
+  More info about them will be provided in next sections.
+- debug metrics. 
+  They are turned off by default, but can be turned on by adding `METRICS_COLLECTOR_TYPE = 'kv'` to `/etc/indy/indy_config.py` and restarting node.
+  Tools for processing these metrics are scripts `process_logs` and `build_graph_from_csv` bundled with Indy Node.
+  Debug metrics can be used to find some insidios problems like memory leaks and some other hard to detect problems.
+
+## Troubleshooting checklist
+
```

Comment:
I would also mention the following:
1) Check that a node is connected to all the nodes it can be connected. If not - check firewall settings, or run <Andrew's ZMQ script>.
2) Check if View Change is not in progress forever 
3) Check that a Node finished catch-up successfully

Also it makes sense to mention that client write requests may be rejected while view change is in progress